### DISCUSSION TITLE
Can do support for integrating click automation script commands? Or how to add custom click commands to fastapi-cli gracefully?

### DISCUSSION THREAD

**Discussion by YaoJusheng** [fastapi/fastapi-cli#104] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
Now I add some custom commands, like:

@click.group()
def cli():
    """FastAPI CLI for running and managing the application."""
    pass

@cli.command()
@click.option("--reload", is_flag=True, default=False, help="Enable auto-reload")
def runserver(reload: bool):
    ...

...
```
```


### Description


In addition, there are some command groups defined by the click library. Currently I can only create other py files, such as `main.py/manage.py`, and then execute `python main.py/manage.py command < options>`.

How to integrate them into fastapi and then run them directly with fastapi command?


### Operating System

macOS

### Operating System Details

macOS 14.6.1 arm64

### Python Version

3.10.12

### Additional Context

_No response_

**Comment by YuriiMotov** [comment]
Could you please explain why you need this?
Why don't just create separate Typer\Click app?

**Reply by YaoJusheng** [reply]
At that time, I noticed that the fastapi-cli library could be used for the simple management of FastAPI applications, yet its configurable options were quite limited. 

It was impossible to extend the CLI without relying on additional library (e.g., Click), however, integrating other CLI dependencies would break its compatibility with FastAPI.

Back then, the idea was to use a unified CLI as much as possible to enable flexible management of app deployment, while also being able to provide potential external extensions through some common interfaces.

Of course, the separated click was used later.

**Reply by YuriiMotov** [reply, marked as answer]
There is no option to configure `fastapi-cli` to add custom commands, but you can always fork the repo and add the functionality you need.
Then, modify you requirements to install FastAPI without `fastapi-cli` (`pip install fastapi` or `pip install fastapi[standard-no-fastapi-cloud-cli]`) dependency and install your modified `fastapi-cli` version instead
